mithril-aggregator/src/artifact_builder/cardano_database.rs

```diff
@@ -0,0 +1,207 @@
+
+        let cardano_database = CardanoDatabase::new(
+            merkle_root.to_string(),
+            beacon,
+            total_db_size_uncompressed,
+            ArtifactsLocations::default(), // TODO: temporary default locations, will be injected in next PR.
+            self.compression_algorithm,
+            &self.cardano_node_version,
+        );
+
+        Ok(cardano_database)
+    }
+}
+
+// Return the sum of the files size contained in the subdirectories 'immutable', 'ledger' and 'volatile'.
+fn compute_uncompressed_database_size(db_directory: &Path) -> StdResult<u64> {
+    let subdirs = ["immutable", "ledger", "volatile"];
+
+    let mut total_db_size_uncompressed = 0;
+    for subdir in subdirs {
+        let dir_path = db_directory.join(subdir);
+
+        total_db_size_uncompressed += get_directory_size(&dir_path)
+            .with_context(|| format!("Failed to read metadata for directory: {:?}", dir_path))?;
+    }
+
+    Ok(total_db_size_uncompressed)
+}
+
+fn get_directory_size(path: &Path) -> StdResult<u64> {
+    let entries =
+        std::fs::read_dir(path).with_context(|| format!("Failed to read directory: {:?}", path))?;
+
+    let mut directory_size = 0;
+    for entry in entries {
+        let path = entry
+            .with_context(|| format!("Failed to read directory entry in {:?}", path))?
+            .path();
+
+        if path.is_file() {
+            let metadata = std::fs::metadata(&path)
+                .with_context(|| format!("Failed to read metadata for file: {:?}", path))?;
+            directory_size += metadata.len();
+        } else if path.is_dir() {
+            directory_size += get_directory_size(&path)?;
+        }
+    }
+
+    Ok(directory_size)
+}
+
+#[cfg(test)]
+mod tests {
+    use std::path::PathBuf;
+
+    use mithril_common::{
+        digesters::DummyImmutablesDbBuilder,
+        entities::{ProtocolMessage, ProtocolMessagePartKey},
+        test_utils::{fake_data, TempDir},
+    };
+
+    use super::*;
+
+    fn get_test_directory(dir_name: &str) -> PathBuf {
+        TempDir::create("cardano_database", dir_name)
+    }
+
+    #[test]
+    fn should_compute_the_size_of_the_uncompressed_database_only_immutable_ledger_and_volatile() {
+        let test_dir = get_test_directory("should_compute_the_size_of_the_uncompressed_database_only_immutable_ledger_and_volatile");
+
+        DummyImmutablesDbBuilder::new(test_dir.as_os_str().to_str().unwrap())
+            .with_immutables(&[1, 2])
+            .set_immutable_file_size(1000)
+            .with_ledger_files(vec!["blocks-0.dat".to_string()])
+            .set_ledger_file_size(5000)
+            .with_volatile_files(vec!["437".to_string(), "537".to_string()])
+            .set_volatile_file_size(2000)
+            .build();
+        // Number of immutable files = 2 × 3 ('chunk', 'primary' and 'secondary').
+        let expected_total_size = 2 * 3 * 1000 + 5000 + 2000 * 2;
```

Comment:
We can add some parenthesis to help understanding the calculation.
```
let expected_total_size = (2 * 3 * 1000) + 5000 + (2 * 2000);
```
We may can use variables to define size so it's more explicit in formula.

It also be interesting to use less ambiguous values.
Here we have 2 immutables of 1000, so a size of 2000 for chunks, primaries and secondaries. 
There is also 2 volatile files of 2000.

Reply:
Is it ok for you like this?
```
        DummyImmutablesDbBuilder::new(test_dir.as_os_str().to_str().unwrap())
            .with_immutables(&[1, 2])
            .set_immutable_file_size(777)
            .with_ledger_files(vec!["blocks-0.dat".to_string()])
            .set_ledger_file_size(6666)
            .with_volatile_files(vec!["437".to_string(), "537".to_string()])
            .set_volatile_file_size(99)
            .build();
        // Number of immutable files = 2 × 3 ('chunk', 'primary' and 'secondary').
        let expected_total_size = (2 * 3 * 777) + 6666 + (2 * 99);
```

Reply:
The test was simplified based on the different comments on the PR: https://github.com/input-output-hk/mithril/pull/2159/files#diff-aab140d744a307dd040e3dba02956b21858cb54510ea96a58ab094124897da8eR116-R128